framework/include/dgkernels/DGInterface.h

```diff
@@ -0,0 +1,56 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef DGINTERFACE_H
+#define DGINTERFACE_H
+
+#include "DGKernel.h"
+
+//Forward Declarations
+class DGInterface;
+
+template<>
+InputParameters validParams<DGInterface>();
+
+/* Objects whose type inherits from DGInterface must currently define the master variable to be
+   the variable whose element ID will be less than the element ID of the neighboring variable.
+   The reason for this can be seen by looking at the ComputeResidualThread::onInterface method */
+
+class DGInterface : public DGKernel
+{
+public:
+  DGInterface(const InputParameters & parameters);
+
+  std::vector<dof_id_type> _connected_dof_indices;
```

Comment:
Do you intend for `_connected_dof_indices`, `_Kne`, and `_Kee` to be public?  They are named in the way we usually name protected variables.